internal/common/predicate.go

```diff
@@ -12,3 +14,14 @@ func IsControlledBy(object client.Object, owner *metav1.OwnerReference) bool {
 		controller.APIVersion == owner.APIVersion &&
 		controller.Kind == owner.Kind
 }
+
+// FetchCurrentNamespaceName returns namespace name by using information stored as file
+// Returns default Openshift Builds namespace on error
+// Refer: https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#without-using-a-proxy
+func FetchCurrentNamespaceName() string {
```

Comment:
:thinking: Since this thing is always running in a Kubernetes pod with a service account + token, I wonder if we should have this return error if we can't read the file.

I think we can merge this as is and discuss once it lands.

Reply:
I think we should have a fallback logic for fetching namespace.